### DISCUSSION TITLE
What is the best way to change the repository name of a formula?

### DISCUSSION THREAD

**Discussion by dehanj** [Homebrew/discussions#5031] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
n/a
```

### Output of `brew doctor`

```shell
n/a
```

### Description of issue

Hi, 
We would like to change a repository name, but I do have a question regarding how that affects [our Formula in Hombrew-core](https://github.com/Homebrew/homebrew-core/blob/master/Formula/t/tkey-ssh-agent.rb). 

Some context is that it from the beginning was a mono-repo (don't ask me why), but we have cleaned it up and now only has one application left (the same as the one you get when you run brew install today) and hence would like to change the repository name to something that is a bit more descriptive of the application. 

We obviously have to make a PR, but my question is what happens between the moment we change the repository name and submit the PR, until the PR is merged?

Will an attempt to install the formula simply return an error that the formula is not found? Or is there any smart way of doing this? 
Thanks for any input.

**Comment by SMillerDev** [comment, marked as answer]
All formulae in Homebrew/core are prebuilt binaries so everything will continue as normal unless someone tries to build from source. But even then GitHub will automatically redirect renamed repositories for a while.

**Comment by ZhongRuoyu** [comment]
Making a PR is the way to change this.

> what happens between the moment we change the repository name and submit the PR, until the PR is merged?

Users should not be affected even before the PR is merged, because [GitHub's own redirect logic](https://docs.github.com/en/repositories/creating-and-managing-repositories/renaming-a-repository) will take care of the rename as long as you don't create another repo with the same name as the old one. (For example, https://github.com/Homebrew/homebrew directs to https://github.com/Homebrew/legacy-homebrew and Git operations on both locations are still valid.) In most cases nothing will be fetched from the source repository anyway, because the bottles (pre-built binaries) will be used if Homebrew is running in a supported configuration.

**Comment by dehanj** [comment]
Thanks for the quick response!
Then we will rename the repo and get a PR going.
